### PR TITLE
Add recently graded assignments to widget

### PR DESCRIPTION
**Pull request by shichongrui**
refs: MBL-12172
affects: Student
release note: Added recently graded assignments to the today widget

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please run `yarn update-headers`. The following do not have the correct license header: 
* Core/CoreTests/API/APIRecentlyGradedRequestableTests.swift
* Core/CoreTests/Use Cases/GetRecentlyGradedSubmissionsTests.swift
  </td>
    </tr>
  </tbody>
</table>



#### Release Note: 
Added recently graded assignments to the today widget

#### Affected Apps: Student

[MBL-12172](https://instructure.atlassian.net/browse/MBL-12172)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 93.69% | 93.62% | 0.07%
**Student** | 92.17% | 92.17% | 0%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.63% | 92.63% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by shichongrui** [comment]
Test plan is in the ticket.

File: Core/Core/API/APIRecentlyGradedRequestable.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+import Foundation
+
+public struct GetRecentlyGradedSubmissionsRequest: APIRequestable {
+    public typealias Response = [APISubmission]
+
+    let userID: String
+
+    public var path: String {
+        return "users/\(userID)/graded_submissions"
```
Comment (armstrongnate) [inline Core/Core/API/APIRecentlyGradedRequestable.swift]: Nitpick: we've been using `ContextModel` for these sorts of paths.

```swift
let context = ContextModel(.user, id: userID)
return "\(context.pathComponent)/graded_submissions"
```

File: Canvas/GradesWidget/GradesTodayWidgetViewController.swift [hunk 2]
```diff
@@ -128,24 +132,87 @@ class GradesTodayWidgetViewController: UIViewController {
...
+    var ASSIGNMENT_GRADE_ROW_HEIGHT: CGFloat {
+        return UIFont.scaledNamedFont(.medium12).lineHeight + UIFont.scaledNamedFont(.semibold16).lineHeight + VERTICAL_PADDING
+    }
+
+    var COURSE_GRADE_ROW_HEIGHT: CGFloat {
+        return UIFont.scaledNamedFont(.semibold16).lineHeight + VERTICAL_PADDING
+    }
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 2
```
Comment (armstrongnate) [inline Canvas/GradesWidget/GradesTodayWidgetViewController.swift]: Do we have/need empty states for these sections? Does it look funny if the recently graded submissions are empty?

File: Core/Core/API/APIRecentlyGradedRequestable.swift (hunk 1 above)
Comment (shichongrui) [reply Core/Core/API/APIRecentlyGradedRequestable.swift]: Oh I knew about this but didn't think of user as a context. I will swap this.

File: Canvas/GradesWidget/GradesTodayWidgetViewController.swift (hunk 2 above)
Comment (shichongrui) [reply Canvas/GradesWidget/GradesTodayWidgetViewController.swift]: Hmm we may want empty states. I will chat with dave.

**Comment by armstrongnate** [comment]
QA +1 looks good.

File: Canvas/GradesWidget/Info.plist [hunk 3]
```diff
@@ -20,6 +20,11 @@
 	<string>6.3.6</string>
 	<key>CFBundleVersion</key>
 	<string>1</string>
+	<key>NSAppTransportSecurity</key>
+	<dict>
+		<key>NSAllowsArbitraryLoads</key>
```
Comment (vanwagonet) [inline Canvas/GradesWidget/Info.plist]: was this only necessary for testing against local canvas?

File: TestsFoundation/TestsFoundation/Fixtures/API/APIAssignmentFixture.swift [hunk 4]
```diff
@@ -37,3 +37,21 @@ extension APIAssignment: Fixture {
         ]
     }
 }
+
+extension APIAssignmentNoSubmission: Fixture {
```
Comment (vanwagonet) [inline TestsFoundation/TestsFoundation/Fixtures/API/APIAssignmentFixture.swift]: Can you update this to the new way we do API fixtures?

File: Core/Core/Database/CoreData/Models/Assignment.swift [hunk 5]
```diff
@@ -152,6 +152,41 @@ extension Assignment {
 
         return utis
     }
+
+    static let scoreFormatter: NumberFormatter = {
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .percent
+        formatter.decimalSeparator = "."
```
Comment (vanwagonet) [inline Core/Core/Database/CoreData/Models/Assignment.swift]: won't this be wrong in some locales?

File: Canvas/GradesWidget/Info.plist (hunk 3 above)
Comment (shichongrui) [reply Canvas/GradesWidget/Info.plist]: Yes

**Comment by vanwagonet** [comment]
I seem to always get an `Unable to Load` message in the widget.

**Comment by vanwagonet** [comment]
ok, I was using a prod account. When I use beta it works!

**Pull request merged**